buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -835,9 +838,9 @@ class BuildPlugin implements Plugin<Project> {
                 test.jvmArgs "-Xmx${System.getProperty('tests.heap.size', '512m')}",
                         "-Xms${System.getProperty('tests.heap.size', '512m')}",
                         '-XX:+HeapDumpOnOutOfMemoryError',
-                        "-XX:HeapDumpPath=$heapdumpDir",
                         '--illegal-access=warn'
 
+                test.jvmArgumentProviders.add({ ["-XX:HeapDumpPath=$heapdumpDir"] } as CommandLineArgumentProvider)
```

Comment:
Can we keep `-XX:+HeapDumpOnOutOfMemoryError` next to this or with it in the provider?

Reply:
It's doesn't really matter, we can move it here if we think it makes sense. Although it carries the implication that it _has_ to be passed this way.

Reply:
It would be nice to keep them next to each other, since they are related to each other.

Reply:
Done.